api/ratelimiter.go

```diff
@@ -25,10 +25,10 @@ func rateLimit(ctx context.Context, limiter limiter.Store, logger zerolog.Logger
 		return nil // if no client identifier disable limit
 	}
 
-	_, _, _, ok, _ := limiter.Take(ctx, remote)
+	requestsLimit, _, _, ok, _ := limiter.Take(ctx, remote)
 	if !ok {
 		logger.Debug().Str("origin", remote).Msg("rate limit reached")
-		return errs.ErrRateLimit
+		return errs.NewRateLimitError(requestsLimit)
```

Comment:
nit: I wouldn't share rate limit with the user, this is a "security concern" since then they will know exactly what the limit is.

Reply:
But does this give any malicious actor an advantage? They still can't send more than the configured limit. They can send slightly less, but we do not expose the time-frame where the limit refreshes.
On the other hand, for some dApps connecting to a Gateway, it might be useful if they know what is the maximum limit of requests.

Reply:
Removed in https://github.com/onflow/flow-evm-gateway/pull/453/commits/6a1589265f5ff617900784a9f746447476d426a8, just in case :pray: